Formula/starfig.rb

```diff
@@ -0,0 +1,15 @@
+class Starfig < Formula
+  desc "Starfig is a programmatic config generator. It helps create static configs using Starlark, a deterministic Python-like language."
+  homepage "https://github.com/jathu/starfig"
+  url "https://github.com/jathu/starfig/releases/download/0.1/starfig-0.1-macos-x86.zip"
+  sha256 "bf254452452b955cb70d95e4ae90b7242f4b811ed4c920fcf7acfe03a3782100"
+  license "Apache-2.0"
+
+  def install
+    bin.install "starfig"
+  end
+
+  test do
+    assert_match("starfig-0.1", shell_output("#{bin}/starfig version"))
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.